Project EFT/Data Classes/StandardUser.cs

```diff
@@ -43,5 +43,45 @@ public void UpdateRanking()
             int newRanking = DBConnector.GetUserRanking(Id);
             if (newRanking != DBConnector.DB_FAILURE) Ranking = newRanking;
         }
+
+        public bool IsEqualWithSubMap(StandardUser otherUser)
+        {
+            bool subMapEqual = true;
+            bool shouldBreak = false;
+
+            //checks the equality of the values in the submission dictionary....
+            foreach (KeyValuePair<int, List<AnswerSubmission>> kv in otherUser.Submissions)
+            {
+
+                if (this.Submissions.ContainsKey(kv.Key))
+                {
+                    int count = 0;
+                    foreach (AnswerSubmission a in kv.Value)
+                    {
+                        if (!this.Submissions[kv.Key][count].IsEqual(a))
+                        {
+                            subMapEqual = false;
+                            shouldBreak = true;
+                        }
+                        count++;
+                    }
+                }
+                else if (shouldBreak)
+                {
+                    break;
+                }
+                else
+                {
+                    subMapEqual = false;
+                    break;
+                }
+            }
+            return (this.Username.Equals(otherUser.Username) && this.Password.Equals(otherUser.Password) && this.Email.Equals(otherUser.Email) && this.Id == otherUser.Id && subMapEqual);
+        }
```

Comment:
paired with the above comment, this could be simplified to just return the result of `IsEqual` as execution would only reach this point if the submissions are known to be equal

Reply:
Wouldn't that add an extra function call, making it more complex memory wise?
For now I left it the way it is, minus the ```&&  subMapEqual``` boolean...